oc-chef-pedant/spec/api/compliance_proxy_spec.rb

```diff
@@ -1,43 +1,73 @@
+def wait_for_http_listener(uri_string, retries = 10)
+  require 'net/http'
+  require 'openssl'
+  require 'uri'
+  uri = URI(uri_string)
+  begin
+    puts "Trying GET #{uri_string}"
+    Net::HTTP.start(uri.host, uri.port, :use_ssl => true,
+                    verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+      request = Net::HTTP::Get.new uri
+      http.request request
+    end
+    puts "Succesfully made GET to #{uri_string}"
+  rescue StandardError => e
+    if retries <= 0
+      raise e
+    else
+      puts "GET #{uri_string} failed: #{e} retrying in 1 seconds (#{retries} retries remaining)"
+      retries -= 1
+      sleep 1
```

Comment:
Minor, but we could use `sleep 0.1` here - if we poll a couple of extra times, no harm done.